src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs

```diff
@@ -171,14 +172,14 @@ private static bool CheckScoreColumnKind(Schema schema, int col)
 
             if (!string.IsNullOrWhiteSpace(name))
             {
-                ColumnInfo info;
 #pragma warning disable MSML_ContractsNameUsesNameof
-                if (!ColumnInfo.TryCreateFromName(schema, name, out info))
+                var col = schema.GetColumnOrNull(name);
+                if (!col.HasValue)
```

Comment:
Sorry, not sure what's is being asked here @sfilipi , could you clarify? `schema` is checked for null, the `name` has been checked for null, and as you see below the nullable struct is being checked for a value and throwing an appropriate exception. As far as I can tell no other entities are involved in the line you're commenting on?

Reply:
I mean col can be null, since it comes from GetColumnOrNull. 

Shall it check for null before accessing members on it: 

Next line: col?.HasValue

---
In reply to: [243391027](https://github.com/dotnet/machinelearning/pull/1924#discussion_r243391027) [](ancestors = 243391027)